src/drivers/amd/common/acp_dmic_dma.c

```diff
@@ -106,10 +107,40 @@ static int acp_dmic_dma_status(struct dma_chan_data *channel,
 static int acp_dmic_dma_copy(struct dma_chan_data *channel, int bytes,
 			     uint32_t flags)
 {
+	uint32_t i;
+	uint32_t j;
+	int numsamples;
+	char *dmic_rngbuff_addr2 = acp_initsilence.dmic_rngbuff_addr1;
 	struct dma_cb_data next = {
 		.channel = channel,
 		.elem.size = bytes,
 	};
+	if (acp_initsilence.silence_incr < acp_initsilence.silence_cnt) {
+		if (acp_initsilence.silence_incr % 2)
+			dmic_rngbuff_addr2 = acp_initsilence.dmic_rngbuff_addr1 +
+								(dmic_rngbuff_size >> 1);
+		for (i = 0; i < (dmic_rngbuff_size >> 1); i = i + 1)
+			dmic_rngbuff_addr2[i] = 0;
+		acp_initsilence.silence_incr++;
+	} else if (acp_initsilence.silence_incr < acp_initsilence.silence_cnt +
+			acp_initsilence.numfilterbuffers) {
+		numsamples = (dmic_rngbuff_size >> 1 / (acp_initsilence.num_chs *
```

Comment:
ouch, no, your previous version had `((dmic_rngbuff_size >> 1) / (acp_initsilence.num_chs *` here and removing those first parentheses *changes* the calculation order - division has higher precedence than a bit-shift in C.

Reply:
Thanks @lyakh. Updated the changes